File: content/atlas/source/includes/fts/field-types/embeddedDocuments/create-index-basic.py
from pymongo import MongoClient
from pymongo.operations import SearchIndexModel

# Connect to your Atlas deployment
uri = "<connection-string>"
client = MongoClient(uri)

database = client["sample_training"]
collection = database["companies"]

# Create the MongoDB Search index definition for the embeddedDocuments field
search_index_model = SearchIndexModel(
    definition={
        "mappings": {
            "dynamic": False,
            "fields": {
                "products":{
                   "type": "embeddedDocuments",
                   "dynamic": True
                }
            }
        },
    },
    name="default",
)

# Create the index
result = collection.create_search_index(model=search_index_model)
print(f"New index name: {result}")

File: content/search/source/includes/index/field-types/autocomplete/code-snippets/java/create_index_basic.java
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class CreateIndex {
    public static void main(String[] args) {
        // connect to your Atlas cluster
        String uri = "<connection-string>";

        try (MongoClient mongoClient = MongoClients.create(uri)) {
            // set namespace
            MongoDatabase database = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> collection = database.getCollection("movies");
            String indexName = "default";

            Document searchIdx = new Document("mappings",
                    new Document("dynamic", false)
                        .append("fields", new Document("title", 
                                              new Document("type", "autocomplete")
                                                  .append("analyzer", "lucene.standard")
                                                  .append("tokenization", "edgeGram")
                                                  .append("minGrams", 3)
                                                  .append("maxGrams", 5)
                                                  .append("foldDiacritics", false)
                                                  .append("similarity", new Document("type", "stableTfl"))
                                                  )));

            collection.createSearchIndex(indexName, searchIdx);
            System.out.println("New index name: " + indexName);
        } catch (Exception e) {
            System.err.println("Error creating Atlas Search index: " + e.getMessage());
        }
    }
}